tests/e2e/vmc_scale_gc_workers.go

```diff
@@ -0,0 +1,324 @@
+					svcPVCName := volumeID
+					volumeID = getVolumeIDFromSupervisorCluster(svcPVCName)
+					gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+					verifyCRDInSupervisor(ctx, f, sspod.Spec.NodeName+"-"+svcPVCName,
+						crdCNSNodeVMAttachment, crdVersion, crdGroup, true)
+
+					ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s",
+						pv.Spec.CSI.VolumeHandle, sspod.Spec.NodeName))
+					var vmUUID string
+					vmUUID, err = getVMUUIDFromNodeName(sspod.Spec.NodeName)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volumeID, vmUUID)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					gomega.Expect(isDiskAttached).To(gomega.BeTrue(),
+						fmt.Sprintf("Volume is not attached to the node, %s", vmUUID))
+
+					ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s",
+						pv.Spec.CSI.VolumeHandle, sspod.Spec.NodeName))
+					verifyIsAttachedInSupervisor(ctx, f, sspod.Spec.NodeName+"-"+pv.Spec.CSI.VolumeHandle,
+						crdVersion, crdGroup)
+
+				}
+			}
+		}
+
+		ginkgo.By("scaling up TKG worker node")
+		if vmcUser == "testuser" {
+			ginkgo.By("Get WCP session id")
+			gomega.Expect((e2eVSphere.Config.Global.VmcDevopsUser)).NotTo(gomega.BeEmpty(), "Devops user is not set")
+			wcpToken = getWCPSessionId(vmcWcpHost, e2eVSphere.Config.Global.VmcDevopsUser,
+				e2eVSphere.Config.Global.VmcDevopsPassword)
+			framework.Logf("vmcWcpHost %s", vmcWcpHost)
+		} else {
+			ginkgo.By("Get WCP session id")
+			gomega.Expect((e2eVSphere.Config.Global.VmcCloudUser)).NotTo(gomega.BeEmpty(), "Cloud user is not set")
+			wcpToken = getWCPSessionId(vmcWcpHost, e2eVSphere.Config.Global.VmcCloudUser,
+				e2eVSphere.Config.Global.VmcCloudPassword)
+			framework.Logf("vmcWcpHost %s", vmcWcpHost)
+			wcpToken = getWCPSessionId(vmcWcpHost, e2eVSphere.Config.Global.VmcCloudUser, e2eVSphere.Config.Global.VmcCloudPassword)
+		}
+
+		scaleTKGWorker(vmcWcpHost, wcpToken, "test-cluster-e2e-script", 5)
+		defer func() {
+			if vmcUser == "testuser" {
+				ginkgo.By("Get WCP session id")
+				gomega.Expect((e2eVSphere.Config.Global.VmcDevopsUser)).NotTo(gomega.BeEmpty(), "Devops user is not set")
+				wcpToken = getWCPSessionId(vmcWcpHost, e2eVSphere.Config.Global.VmcDevopsUser,
+					e2eVSphere.Config.Global.VmcDevopsPassword)
+				framework.Logf("vmcWcpHost %s", vmcWcpHost)
+			} else {
+				ginkgo.By("Get WCP session id")
+				gomega.Expect((e2eVSphere.Config.Global.VmcCloudUser)).NotTo(gomega.BeEmpty(), "Devops user is not set")
+				wcpToken = getWCPSessionId(vmcWcpHost, e2eVSphere.Config.Global.VmcCloudUser,
+					e2eVSphere.Config.Global.VmcCloudPassword)
+				framework.Logf("vmcWcpHost %s", vmcWcpHost)
+				wcpToken = getWCPSessionId(vmcWcpHost, e2eVSphere.Config.Global.VmcCloudUser, e2eVSphere.Config.Global.VmcCloudPassword)
+			}
+			scaleTKGWorker(vmcWcpHost, wcpToken, "test-cluster-e2e-script", 2)
```

Comment:
why replicas are 2 here

Reply:
That's the test case I want to validate, after scaling up the tkg worker and sts pods, ill scale down the tkg worker nodes, post scaling down the tkg worker nodes all the sts pods should be reattached to the available nodes